Drivers/Src/LTC6811.c

```diff
@@ -138,6 +138,125 @@ void LTC6811_Init(cell_asic *battMod){
 	LTC6811_init_reg_limits(NUM_MINIONS, battMod);
 }
 
+// Reads and parses the LTC6811 cell voltage registers. If packets fail to be read correctly too many times in a row, the BPS is tripped.
+void LTC6811_rdcv_wrap(uint8_t reg, // Controls which cell voltage register is read back.
+                     uint8_t total_ic, // the number of ICs in the system
+                     cell_asic ic[] // Array of the parsed cell codes
+                    )
+{
+  OS_ERR err;
+  uint8_t count = 0;
+  while(LTC681x_rdcv(reg,total_ic,ic) > -1){
+    if(count == MAX_PEC_ERRORS){
+      //trip BPS
+      OSSemPost(&Fault_Sem4, OS_OPT_POST_1, &err);
+    }
+    count++;
+  }
+}
```

Comment:
I think this will fault unless we drop `MAAX_PEC_ERRORS` in a row. Shouldn't it be something like
`while(LTC6811xrdcv(reg,total_ic,ic) != 0){`
 or
`while(LTC6811xrdcv(reg,total_ic,ic) == -1){`